android/app/src/main/res/layout-land/layout_nav_top.xml

```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+  xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  xmlns:tools="http://schemas.android.com/tools"
+  android:id="@+id/nav_top_frame"
+  android:layout_width="match_parent"
+  android:layout_height="match_parent"
+  tools:background="#20FF0000"
+  tools:showIn="@layout/layout_nav">
+  <FrameLayout
+    android:id="@+id/street_frame"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:elevation="@dimen/nav_elevation"
+    app:layout_constraintTop_toTopOf="parent"
+    android:clickable="true"
+    android:background="?cardBackground">
+    <RelativeLayout
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      tools:ignore="UselessParent">
+      <TextView
+        android:id="@+id/street"
+        style="@style/MwmWidget.TextView.NavStreet"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/nav_street_left"
+        android:maxLines="2"
+        android:layout_gravity="center_horizontal"
+        android:gravity="center"
+        tools:text="Sample street name.\nLong looooooooong!!!!"/>
+    </RelativeLayout>
+  </FrameLayout>
+
+  <RelativeLayout
+    android:id="@+id/nav_next_turn_container"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    app:layout_constraintTop_toBottomOf="@id/street_frame"
+    app:layout_constraintStart_toStartOf="parent"
+    android:layout_marginTop="-40dp"
```

Comment:
Should it be hard-coded or made relative in some other way? How was this value determined? Will it work with a very large font set in system settings?

Reply:
This file was just copied from https://github.com/organicmaps/organicmaps/blob/656686cbb4f768d0c28313e7921aa555992a68eb/android/app/src/main/res/layout/layout_nav_top.xml, there are only three lines that are different, and they're in the first FrameLayout section:
```
~ $ diff layout_nav_top.xml layout_land_nav_top.xml 
21c21
<       android:layout_height="@dimen/nav_street_height"
---
>       android:layout_height="wrap_content"
27c27
<         android:layout_height="match_parent"
---
>         android:layout_height="wrap_content"
30c30
<         android:layout_gravity="center_vertical"
---
>         android:layout_gravity="center_horizontal"
```
I don't remember the exact reason I changed them, I think it was based on some of the requests to make landscape look different to portrait mode. Personally I think changes to landscape might fit better in a different PR (and would be better off being done by someone who understands the android layout xml better), but that was what was requested so I figured out how to make it look different